src/data_provider/src/packages/packageLinuxParserHelper.h

```diff
@@ -133,6 +135,113 @@ namespace PackageLinuxHelper
         return ret;
     }
 
+
+    static nlohmann::json parseSnap(const nlohmann::json& info)
+    {
+        nlohmann::json ret;
+
+        std::string name;
+        std::string version;
+        std::string vendor;
+        std::string install_time;
+        std::string location;
+        std::string architecture;
+        std::string groups;
+        std::string description;
+        std::string priority;
+        std::string multiarch;
+        std::string source;
+        std::string format;
+        int         size = 0;
+
+        bool        hasName    = false;
+        bool        hasVersion = false;
+
+        // TODO: Dismiss record if name or version are not present. 
+
+        if (info.contains("name")) 
+        {
+            name = info.at("name");
+            
+            if (name.length()) 
+            {
+                hasName = true;
+            }
+        } 
+
+        if (info.contains("version")) 
+        {
+            version = info.at("version");
+
+            if (version.length()) 
+            {
+                hasVersion = true;
+            }
+        }
+
+        if (!(hasVersion && hasName)) {
+            ret.clear();
+            return ret;
+        }
+
+        if (info.contains("publisher")) 
+        {
+            auto &publisher = info.at("publisher");
+            
+            if (publisher.contains("display-name")) 
+            {
+                vendor = publisher.at("display-name");
+            }
+        }
+
+        if (info.contains("install-date")) 
+        {
+            install_time = info.at("install-date");
+        }
+
+        if (info.contains("summary")) 
+        {
+            description = info.at("summary");
+        }
+
+        if (info.contains("installed-size")) 
+        {
+            auto &data = info.at("installed-size");
+
+            if (data.is_number()) 
+            {
+                size = data.get<int>();
+            } 
+            else if (data.is_string()) 
+            {
+                auto stringData = data.get_ref<const std::string&>();
+
+                if (stringData.length()) 
+                {
+                    size = std::stoi( stringData );
+                }
+            }
+        }
+
+        ret["name"]             = name;
+        ret["location"]         = "/snap/" + name;
+        ret["version"]          = version;
+        ret["vendor"]           = vendor;
```

Comment:
In case some of the values weren't found, we should insert `UNKNOWN_VALUE`. This section assumes that there is always a vendor, description, etc. field.